data/transactions/logic/assembler.go

```diff
@@ -1857,10 +1857,13 @@ func splitTokens(tokens []string) (current, rest []string) {
 
 // assemble reads text from an input and accumulates the program
 func (ops *OpStream) assemble(text string) error {
-	fin := strings.NewReader(text)
+	if strings.TrimSpace(text) == "" {
+		return ops.errorf("Cannot assemble empty program text")
+	}
```

Comment:
```suggestion
	// version check
	...
	trimmed := strings.TrimSpace(text)
	if trimmed == "" {
		return ops.errorf("Cannot assemble empty program text")
	}
	fin := strings.NewReader(trimmed)
```

Reply:
Implemented suggestion in 5f7de10984475128b8d2ec5871e10318f844fd8e

Reply:
I am reverting this change. Though I agree that passing on `trimmed` seems sensible, it is breaking tests, and the asserted behavior is there for a reason. For example, when providing this TEAL input:
```

int 1
#pragma version 2
```

[This](https://github.com/algorand/go-algorand/blob/5f7de10984475128b8d2ec5871e10318f844fd8e/data/transactions/logic/assembler_test.go#L2309) test case is failing:

This is because it is expecting the error:
```go
{l: 3, s: "#pragma version is only allowed before instructions"}
```

but now, since we removed the empty line, it becomes
```go
{l: 2, s: "#pragma version is only allowed before instructions"}
```

From a user's perspectie, I believe the original error is more correct.